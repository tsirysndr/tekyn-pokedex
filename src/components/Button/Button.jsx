import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "@emotion/native";
import DefaultText from "@Components/Text";
import { useFonts, Roboto_700Bold } from "@expo-google-fonts/roboto";

const Container = styled.TouchableOpacity`
  background-color: ${(props) => props.theme.colors[props.color]};
  border-radius: 20px;
  width: 120px;
  height: 60px;
  justify-content: center;
  align-items: center;
`;

const Text = styled(DefaultText)`
  font-size: 14px;
  ${({ fontsLoaded }) =>
    fontsLoaded &&
    css`
      font-family: "Roboto_700Bold";
    `}
`;

export const ButtonText = ({ children }) => {
  const [fontsLoaded] = useFonts({ Roboto_700Bold });
  return <Text fontsLoaded>{children}</Text>;
};

export default function Button(props) {
  return <Container {...props}>{props.children}</Container>;
}

Button.defaultProps = {
  children: null,
  color: "default",
};

Button.propTypes = {
  children: PropTypes.node,
  onPress: PropTypes.func,
  color: PropTypes.oneOf([
    "default",
    "normal",
    "fire",
    "water",
    "grass",
    "electric",
    "electrique",
    "ice",
    "fight",
    "fighting",
    "poison",
    "ground",
    "flight",
    "flying",
    "psychic",
    "bug",
    "rock",
    "ghost",
    "dragon",
    "dark",
    "steel",
    "fairy",
  ]),
};
