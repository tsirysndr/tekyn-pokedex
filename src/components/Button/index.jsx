import Button, { ButtonText } from './Button';

export { ButtonText };

export default Button;