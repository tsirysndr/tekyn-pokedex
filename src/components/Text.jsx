import React from "react";
import PropTypes from "prop-types";
import { TouchableHighlight } from "react-native";
import styled from "@emotion/native";
import { useFonts, Roboto_500Medium } from "@expo-google-fonts/roboto";

const StyledText = styled.Text`
  color: ${props => props.color};
  font-family: 'Roboto_500Medium';
`;

export default function Text({ color, onPress, children }) {
  const [fontsLoaded] = useFonts({ Roboto_500Medium });
  if (!fontsLoaded) {
    return null;
  }
  return <StyledText color={color}>{children}</StyledText>;
}

Text.defaultProps = {
  children: null,
  color: "#fff",
};

Text.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
};
