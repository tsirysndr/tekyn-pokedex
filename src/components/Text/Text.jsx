import React from "react";
import PropTypes from "prop-types";
import { TouchableHighlight } from "react-native";
import styled, { css } from "@emotion/native";
import { useFonts, Roboto_500Medium } from "@expo-google-fonts/roboto";

const StyledText = styled.Text`
  color: ${(props) => props.color};
  ${({ fontsLoaded }) =>
    fontsLoaded &&
    css`
      font-family: "Roboto_500Medium";
    `}
`;

export default function Text(props) {
  const [fontsLoaded] = useFonts({ Roboto_500Medium });
  return (
    <StyledText {...props} fontsLoaded={fontsLoaded}>
      {props.children}
    </StyledText>
  );
}

Text.defaultProps = {
  children: null,
  color: "#fff",
};

Text.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
};
