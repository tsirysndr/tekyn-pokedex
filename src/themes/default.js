export const theme = {
  // Colors
  colors: {
   normal: "#A8A87B",
   fire: "#EE803B",
   water: "#6A92ED",
   grass: "#7BC757",
   electrique: "#F7CF43",
   ice: "#9AD8D8",
   fight: "#BE322E",
   poison: "#9B4A99",
   ground: "#DFBF6E",
   flight: "#A893ED",
   psychic: "#EC5B89",
   bug: "#A8B732",
   rock: "#B89F41",
   ghost: "#705A97",
   dark: "#705849",
   dragon: "#7043F4",
   steel: "#B8B9CF",
   fairy: "#EFB7BD",
  }
}