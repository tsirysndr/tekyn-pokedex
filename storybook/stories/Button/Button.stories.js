import { action } from "@storybook/addon-actions";
import { text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react-native";
import React from "react";
import Button, { ButtonText } from "@Components/Button";
import Text from "@Components/Text";
import CenterView from "../CenterView";
import { ThemeProvider } from "@emotion/react";
import { theme } from "@Themes/default";

storiesOf("Button", module)
  .addDecorator((getStory) => (
    <ThemeProvider theme={theme}>{getStory()}</ThemeProvider>
  ))
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("Normal", () => (
    <>
      <Button onPress={action("clicked-text")} color='normal'>
        <ButtonText>{text("Button text", "Normal")}</ButtonText>
      </Button>
    </>
  ))
  .add("Fire", () => (
    <>
      <Button onPress={action("clicked-text")} color='fire'>
        <ButtonText>{text("Button text", "Fire")}</ButtonText>
      </Button>
    </>
  ))
  .add("Water", () => (
    <>
      <Button onPress={action("clicked-text")} color='water'>
        <ButtonText>{text("Button text", "Water")}</ButtonText>
      </Button>
    </>
  ))
  .add("Grass", () => (
    <>
      <Button onPress={action("clicked-text")} color='grass'>
        <ButtonText>{text("Button text", "Grass")}</ButtonText>
      </Button>
    </>
  ))
  .add("Electrique", () => (
    <>
      <Button onPress={action("clicked-text")} color='electrique'>
        <ButtonText>{text("Button text", "Electrique")}</ButtonText>
      </Button>
    </>
  ))
  .add("Ice", () => (
    <>
      <Button onPress={action("clicked-text")} color='ice'>
        <ButtonText>{text("Button text", "ice")}</ButtonText>
      </Button>
    </>
  )).add("Fight", () => (
    <>
      <Button onPress={action("clicked-text")} color='fight'>
        <ButtonText>{text("Button text", "Fight")}</ButtonText>
      </Button>
    </>
  ))
  .add("Poison", () => (
    <>
      <Button onPress={action("clicked-text")} color='poison'>
        <ButtonText>{text("Button text", "Poison")}</ButtonText>
      </Button>
    </>
  ))
  .add("Ground", () => (
    <>
      <Button onPress={action("clicked-text")} color='ground'>
        <ButtonText>{text("Button text", "Ground")}</ButtonText>
      </Button>
    </>
  ))
  .add("Flight", () => (
    <>
      <Button onPress={action("clicked-text")} color='flight'>
        <ButtonText>{text("Button text", "Flight")}</ButtonText>
      </Button>
    </>
  ))
  .add("Psychic", () => (
    <>
      <Button onPress={action("clicked-text")} color='psychic'>
        <ButtonText>{text("Button text", "Psychic")}</ButtonText>
      </Button>
    </>
  ))
  .add("Bug", () => (
    <>
      <Button onPress={action("clicked-text")} color='bug'>
        <ButtonText>{text("Button text", "Bug")}</ButtonText>
      </Button>
    </>
  ))
  .add("Rock", () => (
    <>
      <Button onPress={action("clicked-text")} color='rock'>
        <ButtonText>{text("Button text", "Rock")}</ButtonText>
      </Button>
    </>
  ))
  .add("Ghost", () => (
    <>
      <Button onPress={action("clicked-text")} color='ghost'>
        <ButtonText>{text("Button text", "Ghost")}</ButtonText>
      </Button>
    </>
  ))
  .add("Dark", () => (
    <>
      <Button onPress={action("clicked-text")} color='dark'>
        <ButtonText>{text("Button text", "Dark")}</ButtonText>
      </Button>
    </>
  ))
  .add("Dragon", () => (
    <>
      <Button onPress={action("clicked-text")} color='dragon'>
        <ButtonText>{text("Button text", "Dragon")}</ButtonText>
      </Button>
    </>
  ))
  .add("Steel", () => (
    <>
      <Button onPress={action("clicked-text")} color='steel'>
        <ButtonText>{text("Button text", "Steel")}</ButtonText>
      </Button>
    </>
  ))
  .add("Fairy", () => (
    <>
      <Button onPress={action("clicked-text")} color='fairy'>
        <ButtonText>{text("Button text", "fairy")}</ButtonText>
      </Button>
    </>
  ));
