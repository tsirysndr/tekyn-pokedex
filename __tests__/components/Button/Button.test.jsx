import { render, waitFor } from "@testing-library/react-native";
import Button from "@Components/Button";
import { ThemeProvider } from "@emotion/react";
import { theme } from "@Themes/default";
import Text from "@Components/Text";

test("renders default Button", () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button>
        <Text>Default Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders normal button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="fire">
        <Text>Normal Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});


test("renders fire button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="fire">
        <Text>Fire Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});


test("renders water button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="water">
        <Text>Water Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders grass button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="grass">
        <Text>Grass Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders electric button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="electric">
        <Text>Electric Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});


test("renders ice button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="ice">
        <Text>Ice Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders fight button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="fight">
        <Text>Flight Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders poison button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="poison">
        <Text>Poison Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders ground button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="ground">
        <Text>Ground Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders flight button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="flight">
        <Text>Flight Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders psychic button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="psychic">
        <Text>Psychic Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders bug button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="bug">
        <Text>Bug Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders rock button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="rock">
        <Text>Rock Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders ghost button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="ghost">
        <Text>Ghost Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});


test("renders dragon button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="dragon">
        <Text>dragon Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders dark button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="dark">
        <Text>Dark Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("renders steel button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="steel">
        <Text>Steel Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});


test("renders fairy button", async () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <Button color="fairy">
        <Text>Fairy Button</Text>
      </Button>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});




