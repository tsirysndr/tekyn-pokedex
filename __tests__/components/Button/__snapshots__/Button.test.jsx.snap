// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bug button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#A8B732",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Bug Button
  </Text>
</View>
`;

exports[`renders dark button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#705849",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Dark Button
  </Text>
</View>
`;

exports[`renders default Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4A261",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Default Button
  </Text>
</View>
`;

exports[`renders dragon button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7043F4",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    dragon Button
  </Text>
</View>
`;

exports[`renders electric button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7CF43",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Electric Button
  </Text>
</View>
`;

exports[`renders fairy button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EFB7BD",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Fairy Button
  </Text>
</View>
`;

exports[`renders fight button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#BE322E",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Flight Button
  </Text>
</View>
`;

exports[`renders fire button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EE803B",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Fire Button
  </Text>
</View>
`;

exports[`renders flight button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#A893ED",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Flight Button
  </Text>
</View>
`;

exports[`renders ghost button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#705A97",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Ghost Button
  </Text>
</View>
`;

exports[`renders grass button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7BC757",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Grass Button
  </Text>
</View>
`;

exports[`renders ground button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DFBF6E",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Ground Button
  </Text>
</View>
`;

exports[`renders ice button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9AD8D8",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Ice Button
  </Text>
</View>
`;

exports[`renders normal button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EE803B",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Normal Button
  </Text>
</View>
`;

exports[`renders poison button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9B4A99",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Poison Button
  </Text>
</View>
`;

exports[`renders psychic button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EC5B89",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Psychic Button
  </Text>
</View>
`;

exports[`renders rock button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#B89F41",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Rock Button
  </Text>
</View>
`;

exports[`renders steel button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#B8B9CF",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Steel Button
  </Text>
</View>
`;

exports[`renders water button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6A92ED",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 120,
    }
  }
>
  <Text
    color="#fff"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Water Button
  </Text>
</View>
`;
